migrations/1725996635075_add-virtual-third-parties.ts

```diff
@@ -0,0 +1,110 @@
+/* eslint-disable @typescript-eslint/naming-convention */
+import { MigrationBuilder } from 'node-pg-migrate'
+import { VirtualThirdParty } from '../src/ThirdParty/VirtualThirdParty.model'
+import { Collection } from '../src/Collection/Collection.model'
+
+const update_updated_at_column_function_name = 'update_updated_at_column'
+const update_updated_at_trigger_name = 'update_updated_at'
+const delete_virtual_third_party_function_name =
+  'delete_virtual_third_party_function'
+const delete_virtual_third_party_trigger_name =
+  'delete_virtual_third_party_trigger'
+
+export async function up(pgm: MigrationBuilder): Promise<void> {
+  pgm.createTable(VirtualThirdParty.tableName, {
+    id: {
+      type: 'TEXT',
+      primaryKey: true,
+      notNull: true,
+    },
+    managers: {
+      type: 'TEXT',
+      notNull: true,
+    },
+    raw_metadata: {
+      type: 'TEXT',
+      notNull: true,
+    },
+    created_at: {
+      type: 'TIMESTAMP',
+      notNull: true,
+      default: pgm.func('current_timestamp'),
+    },
+    updated_at: {
+      type: 'TIMESTAMP',
+      notNull: true,
+      default: pgm.func('current_timestamp'),
+    },
+  })
+
+  // Create the trigger function to automatically update the updated_at column
+  pgm.createFunction(
+    update_updated_at_column_function_name,
+    [],
+    {
+      returns: 'TRIGGER',
+      language: 'plpgsql',
+    },
+    `
+    BEGIN
+      NEW.updated_at = NOW();
+      RETURN NEW;
+    END;
+    `
+  )
+
+  // Create the trigger to update the updated_at column
+  pgm.createTrigger(
+    VirtualThirdParty.tableName,
+    update_updated_at_trigger_name,
+    {
+      when: 'BEFORE',
+      operation: 'UPDATE',
+      level: 'ROW',
+      function: update_updated_at_column_function_name,
+      condition: 'OLD.* IS DISTINCT FROM NEW.*',
```

Comment:
shouldn't the updated_at field be updated with each update? or should it meet a specific condition?

Reply:
The `updated_at` column will get updated only if any column in the row gets updated to something new. If it doesn't, we'll be saving a write operation by not updating the `updated_at`.